internal/expr/expr_test.go

```diff
@@ -650,17 +717,17 @@ func (s *ExprSuite) TestMapError(c *C) {
 		"all output into map star",
 		"SELECT &M.* FROM person WHERE name = 'Fred'",
 		[]any{sqlair.M{}},
-		"cannot prepare statement: output expression: columns must be specified for non-struct type: &M.*",
+		"cannot prepare statement: output expression: cannot use map with asterisk unless columns are specified: &M.*",
```

Comment:
We sometimes call it `asterisk` and sometimes `star`, I think we should only use one of them.

Reply:
`asterisk` is less ambiguous for me/what I'm used to. But if there's a good reason to use `star` I'm happy to switch.

Reply:
To me both of them look good, let's keep asterisk. But then, change all the places that say `star` in the code.

Reply:
Then we have two concepts for the same things, some error messages and code say one thing and some say the other. Then I would say keep it as `star` and change it to `asterisk` in another PR to avoid having both at the same time.

Reply:
The only mentions of "star" in the codebase are in test names and variable names, all of the variable names it appears in will disappear when we put methods on `valueAccessor`. The test names I'm really not worried about. It doesn't appear anywhere in the error messages. I think its fine just to leave it for now.

Reply:
Ok let's call it a matter of preference. Because it is not in the error messages, let's leave it as is.